core/services/pipeline/task_params_test.go

```diff
@@ -504,6 +506,223 @@ func TestJSONWithVarExprs(t *testing.T) {
 	}
 }
 
+func TestResolveValue(t *testing.T) {
```

Comment:
This is not a new code, just moved these tests from `variables_test.go` to `task_params_test.go`, because these tests belong to task params.